cpp/src/parquet/row_ranges.h

```diff
@@ -0,0 +1,201 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
```

Comment:
overall this doesn't appear to be updated to reflect the interfaces discussed in https://docs.google.com/document/d/1SeVcYudu6uD9rb9zRAnlLGgdauutaNZlAaS0gVzjkgM/edit ?

Reply:
@huberylee Do you mind taking a look at this patch: https://github.com/apache/arrow/pull/38867? It does similar feature but use a different approach discussed in the doc without optimization I/O and would be good to consolidate the effort with yours.

Reply:
> overall this doesn't appear to be updated to reflect the interfaces discussed in https://docs.google.com/document/d/1SeVcYudu6uD9rb9zRAnlLGgdauutaNZlAaS0gVzjkgM/edit ?

I have already started working on it during the documentation discussion, so the implementation in this MR will be a little different from the documentation.

Reply:
> @huberylee Do you mind taking a look at this patch: #38867? It does similar feature but use a different approach discussed in the doc without optimizing I/O and would be good to consolidate the effort with yours.

I'm pleasure to do it. I will look at the patch: #38867 later.